### PR TITLE
Fix C issues found by coverity and others

### PR DESCRIPTION
**Pull request by t-woerner**
Two issues have been addressed:

    Fix ressource leak in daemons/ipa-slapi-plugins/ipa-cldap/ipa_cldap_netlogon.c ipa_cldap_netlogon
    
    The leak happens due to using strndup in a for loop to create a temporary
    string without freeing it in all cases.

and

    Fix ressource leak in client/config.c get_config_entry
    
    The leak happens due to using strndup to create a temporary string without
    freeing it afterwards.

### CODE REVIEW

**Comment by abbra** [comment]
Please split the commits into separate ones.
Please create a single issue at pagure for backporting purposes.

**Comment by t-woerner** [comment]
I have been splitting up the fixes and changed the text, but it has been changed back. There are still issues with pull requests.

File: daemons/ipa-slapi-plugins/ipa-cldap/ipa_cldap_netlogon.c [hunk 1]
```diff
@@ -350,6 +354,7 @@ int ipa_cldap_netlogon(struct ipa_cldap_ctx *ctx,
             ret = EINVAL;
             goto done;
         }
+        free(domain);
```
Comment (tiran) [inline daemons/ipa-slapi-plugins/ipa-cldap/ipa_cldap_netlogon.c]: This will lead to a double-free error. ``domain`` is already freed in the done block.

**Comment by tiran** [comment]
master:

* 389c17c5c6656027fcece6ddd4322e4e64f17ff6 Fix ressource leak in client/config.c get_config_entry
* 8e98c72e5186d099eb4ac0d8b059c4fce64fd739 Fix ressource leak in daemons/ipa-slapi-plugins/ipa-cldap/ipa_cldap_netlogon.c ipa_cldap_netlogon

**Pull request closed without merging**